sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs

```diff
@@ -187,13 +187,13 @@ private static ValueTask ProcessNextAsync(HttpMessage message, ReadOnlyMemory<Ht
 #if NETCOREAPP2_1
             (s_diagnosticSource.IsEnabled() || ActivityExtensions.ActivitySourceHasListeners(s_activitySource));
 #else
-            (s_diagnosticSource.IsEnabled() || s_activitySource.HasListeners());
+            (s_diagnosticSource.IsEnabled() || (s_activitySource.HasListeners() && ActivityExtensions.SupportsActivitySource));
```

Comment:
Can ShouldCreateActivity be simplified to use IsActivitySourceEnabled?

Reply:
updated